src/Microsoft.Azure.WebJobs.Host/Common/ConverterManager.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs
+{
+    // Concrete implementation of IConverterManager
+    internal class ConverterManager : IConverterManager
+    {
+        // Map from <TSrc,TDest> to a converter function. 
+        private Dictionary<string, object> _funcs = new Dictionary<string, object>();
+
+        public ConverterManager()
+        {
+            this.AddConverter<byte[], string>(DefaultByteArray2String);
+        }
+
+        private static string DefaultByteArray2String(byte[] bytes)
+        {
+            string str = Encoding.UTF8.GetString(bytes);
+            return str;
+        }
+
+        private static string GetKey<TSrc, TDest>()
+        {
+            return typeof(TSrc).FullName + "|" + typeof(TDest).FullName;
+        }
+
+        public void AddConverter<TSrc, TDest>(Func<TSrc, TDest> converter)
+        {
+            string key = GetKey<TSrc, TDest>();
```

Comment:
How are conflicts handled (looks like last one wins)? I'm concerned about cases where there are say 5 different extension bindings loaded into an app, and some of they attempt to define the same conversion. Can't that break bindings? E.g., if multiple bindings define a string->T conversions and a binding is relying on that conversion being done a certain way.